opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -786,6 +787,20 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
         }
     }
 
+    void Schedule::updateICDScalingFactors() {
+        // updating the scaling factors for all SICD and AICD segments
+        auto& sched_state = this->snapshots.back();
+        for (const auto& well : sched_state.wells()) {
+            if (!well.get().isMultiSegment()) {
+                continue;
+            }
+            auto new_well{well.get()};
+            if (new_well.updateICDFlowScalingFactors()) {
+                sched_state.wells.update( std::move(new_well) );
```

Comment:
This seems a little scary to me.  You're effectively updating a structure over which you're iterating a sequence of references.  I think I'd probably make this a two-stage process instead
```C++
auto updatedWells = std::vector<Well>{};
for (const auto& wellPair : sched_state.wells) { // Note: .wells rather than .wells()
    if (! wellPair.second->isMultiSegment()) { continue; }

    auto new_well = *wellPair.second;
    if (new_well.updateICDFlowScalingFactors()) {
        updatedWells.push_back(std::move(new_wells));
    }
}

for (auto&& updatedWell : updatedWells) {
    sched_state.wells.update(std::move(updatedWell));
}
```